broker-core/src/main/java/io/zeebe/broker/logstreams/processor/TypedStreamProcessor.java

```diff
@@ -159,16 +162,38 @@ public void wrap(
 
     @Override
     public void processEvent() {
-      writer.reset();
-      responseWriter.reset();
+      try {
+
+        writer.reset();
+        responseWriter.reset();
 
-      this.writer.configureSourceContext(streamProcessorId, position);
+        this.writer.configureSourceContext(streamProcessorId, position);
 
-      // default side effect is responses; can be changed by processor
-      sideEffectProducer = responseWriter;
+        // default side effect is responses; can be changed by processor
+        sideEffectProducer = responseWriter;
 
-      eventProcessor.processRecord(
-          position, event, responseWriter, writer, this::setSideEffectProducer);
+        eventProcessor.processRecord(
+            position, event, responseWriter, writer, this::setSideEffectProducer);
+      } catch (Exception exception) {
+        final String errorMessage =
+            String.format(PROCESSING_ERROR_MESSAGE, event, exception.getMessage());
+        LOG.error(errorMessage, exception);
+
+        if (event.metadata.getRecordType() == RecordType.COMMAND) {
+          sendCommandRejectionOnException(errorMessage);
+        } else if (event.getMetadata().getRecordType() == RecordType.EVENT) {
+          // TODO(#2028) clean up state
```

Comment:
Is this still relevant? Aren't we already rolling back the transaction?

Reply:
It is still, because it is meant to erase the complete instance from the state if the processing is not recoverable, for that we need the distinction between recoverable and non recoverable exception and a retry mechanism on recoverable exceptions.